readme.md
![](thumbnail.png)

# pY Pollution & Energy

Mod features:
- Add pollution values to all entities in the game (assemblers, miners) that previously didn't have them.
- Playing with all of the Pyanodon's mods takes much longer, and on default vanilla settings it is pretty much impossible to play with biters enabled. We added some recommended map presets that play well with our pollution settings.
- Voiding of some fluids and gases should not generate pollution (Oxygen, Water, Hydrogen etc.).
- Raw fuels like Wood, Coal and Crude Oil now pollute more while processed fuels pollute less. This should motivate players to build advanced processing chains.
- Combustion Turbine rebalancing.
- Hydrogen rebalancing - It now takes more energy to produce Hydrogen than it provides but Hydrogen now has 0% pollution modifier (like with real Hydrogen).
- Pollution-absorbing buildings now absorb more pollution when used with advanced recipes (Only when playing without pyAlienLife).
- Solar Panels are now significantly more expensive to compensate for their high value in long games.

## Dependencies

This mod is made to be played with entire pY Suite (except for pY Alien Life), so all these mods are required:
- Pyanodons Raw Ores
- Pyanodons Fusion Energy
- Pyanodons Coal Processing
- Pyanodons HighTech
- Pyanodons Petroleum Handling
- Pyanodons Industry
- Pyanodons AlienLife (optional)

# Diving deeper

## Map presets

![](doc/map-presets.png)

There are 3 map presets available:
- **Easy** - For newbies and people who break under pressure (:D)
    - Pollution factor: 2 (1 with pyAlienLife)
    - Destroy factor: 50
    - Time factor: 0
    - Starting area: 133%
    - Absorbtion modifier: 150%
    - Attack cost modifier: 150%
- **Default**
    - Pollution factor: 3 (2 with pyAlienLife)
    - Destroy factor: 75
    - Time factor: 5 (4 with pyAlienLife)
    - Starting area: 133%
    - Absorbtion modifier: 125%
    - Attack cost modifier: 125%
- **Hard** - For those who like some challenge or for pY professionals, feel free to add more if you feel like it.
    - Pollution factor: 4 (3 with pyAlienLife)
    - Destroy factor: 100
    - Time factor: 10 (8 with pyAlienLife)
    - Starting area: 133%
    - Absorbtion modifier: 100%
    - Attack cost modifier: 100%

## Combustion Turbine

![](doc/combustion-turbine.png)

This has been balanced in a way that it's slightly worse than burning Coal at MK01 and then gets much better at higher MKs.
Basically, Combustion Turbine MK01 produces more energy in comparison to boilers but at a cost of higher pollution.  
We also changed some Combustion Mixture recipes, some were not viable and some were way too strong. Combustion Mixture recipe effectiveness still varies though.

## Fuels

![](doc/boiler.png)

Examples:  
Coal chain:
- **Wood** - 125% pollution
- **Raw Coal** - 105% pollution
- **Coal** - 100% pollution
- **Coke** - 85% pollution

Natural Gas chain:
- **Natural Gas** - 80% pollution
- **Raw Gas** - 60% pollution
- **Refined Natural Gas** - 50% pollution
- **Purified Natural Gas** - 35% pollution
- **Pure Natural Gas** - 25% pollution

FuelRod chain:
- **FuelRod MK01** - 75% pollution
- **FuelRod MK02** - 65% pollution
- **FuelRod MK03** - 53% pollution
- **FuelRod MK04** - 40% pollution
- **FuelRod MK05** - 25% pollution

As you process further, you get less pollution on your fuels (this is not a general rule, we left some of them untouched at 100% pollution).  
We also made Fuel Rods viable, MK01 starts at 75% pollution and goes all the way down to 49% at MK05.  
Some liquid fuels have had their energy capacity reduced slightly.

## Hydrogen

![](doc/hydrogen.png)

Until now you could use Hydrogen as an easy source of fuel, but that is no more. It is known fact that you need more energy to separate hydrogen from oxygen than the hydrogen provides (around 170 MJ to produce 1kg of Hydrogen which has 120 - 140 MJ of energy).  
Also, 1kg of Hydrogen in gas state has ~12m3 so prepare your largest fluid tanks :)  
One unit of hydrogen now has 15kJ (this is calculated from Electrolyzer MK01, so on higher MKs you can be energy-positive) down from 400kJ, but using Hydrogen as fuel no longer generates ANY pollution.  
You can also use Pressured Hydrogen as fuel if you want to save some space (you also get some extra energy value when you do this).

## Pollution-absorbing buildings

**Note: This feature is not available when pyAlienLife is installed**

![](doc/botanical-nursery.png)

Example:
Botanical Nursery MK01 now has base pollution of -10/m
- Basic recipe for Log takes 80 seconds and has pollution modifier 100%
- Making Logs using water takes 60 seconds, but has pollution modifier 116.667% - now the building has pollution -11.67/m
- The best recipe with Fertilizer takes 10 seconds and has 450% pollution modifier - now the building has pollution -45/m

This works for the following buildings:
- Botanical Nursery
- Moondrop Greenhouse
- Kicalk Plantation
- Ulric Corral
- Mukmoux Pasture
- Guar Gum Plantation

Why? Previously you were basically penalized for using more effective recipes, since you needed less buildings, absorbing less pollution.

Note: Amount of products is also taken into consideration.

# Special thanks

To [Pyanodon](https://github.com/pyanodon) for creating this great mod [suite](https://mods.factorio.com/user/pyanodon).  
You can support his work at his [Patreon](https://www.patreon.com/pyanodon/posts).


changelog.txt
---------------------------------------------------------------------------------------------------
Version: 0.4.8
Date: 2020.10.17
  Changes:
    - Glassworks fix now works even when py Alien Life is not present
---------------------------------------------------------------------------------------------------
Version: 0.4.7
Date: 2020.08.09
  Changes:
    - Reduced Coal Briquette's fuel value to 30MJ and increased it's pollution modifier to 110%
    - Phytoplankton Farm's pollution modifier changed to -10/s
    - Powdered Biomass, Fine Powdered Biomass and Dried Biomass now have the same pollution modifier as Biomass
---------------------------------------------------------------------------------------------------
Version: 0.4.6
Date: 2020.08.01
  Bugfix:
    - Fix loading error introduced in v0.4.5
---------------------------------------------------------------------------------------------------
Version: 0.4.5
Date: 2020.08.01
  Bugfix:
    - Fixing some crashes caused by changes in new pY versions
  Changes:
    - Increased pollution of Atomizer to 10/m at MK01
    - Reduced pollution absorbtion of Fastwood Forestry to -15/m at MK01
    - Biomass fuel value is now 1MJ (this has been set by Pyanodon recently)
---------------------------------------------------------------------------------------------------
Version: 0.4.4
Date: 2020.07.27
  Changes:
    - Coke to Carbon Dioxide recipe now takes 5 Coke and turns it into 200 Carbon Dioxide (instead of 2 -> 80) and takes 4 seconds instead of 10 (in HPF-mk01)
    - Combustion Mixture recipes using Biomass have been adjusted to produce the same amounts as when using Coke, these recipes are:
    - Coal Gas, Coal Slurry, Tall Oil, Hydrogen, Syngas, Gasoline, Diborane, Fuel Rods
    - Doubled the amount of both ingredients and products in Wood->Biomass and Logs->Biomass recipes
    - Duration of Biomass recipes reduced to 9 seconds (from 10)
---------------------------------------------------------------------------------------------------
Version: 0.4.3
Date: 2020.07.20
  Changes:
    - Most Biomass recipes have been rebalanced, notably Kicalk and some other exploitable ones (see forum post for a complete list of changes)
---------------------------------------------------------------------------------------------------
Version: 0.4.2
Date: 2020.07.15
  Changes:
    - Reducing crafting cost of Compost Plants (MK1 - MK4) to 50% of the original amount
    - Reducing duration of compost recipes to 10 seconds (30 seconds was way too high considering the size of the building)
---------------------------------------------------------------------------------------------------
Version: 0.4.1
Date: 2020.07.14
  Changes:
    - Reducing Biomass' pollution from 110% to 105% (equal to Raw Coal)
    - Compost Plant's pollution per minute remains the same at all MKs (as opposed to other buildings for which it increases) and has been reduced from 10/m to 5/m
    - Compost Plant's energy consumption has been significantly reduced (down from ~413kW to 50kW at MK01)
    - Increased duration of all composting recipes to 30 seconds (up from 6 seconds in v0.4.0 and 3 seconds default)
    - Ethanol can now be used as fuel
---------------------------------------------------------------------------------------------------
Version: 0.4.0
Date: 2020.07.12
  Changes:
    - Nerfing Biomass which is stupidly strong as a fuel and can be made from basically anything
    - Biomass now pollutes 10% (slightly more than Raw Coal) more and has 2MJ fuel value (down from 4MJ)
    - Compost Plants now generate more pollution
    - All Biomass recipes now take longer to craft (composting takes time, right?)
    - Crafting Biomass from Oil products is now significantly less effective (why is it even possible?)
    - Some recipes like Sap/Wood/Logs/Moss/Seaweed (easily obtainable stuff) to Biomass have been nerfed
---------------------------------------------------------------------------------------------------
Version: 0.3.4
Date: 2020.04.04
  Bugfix:
    - Mod compatibility fix
  Changes:
    - Glassworks' energy consumption reduced to 1MW (from 27MW) - this fix is enabled by default and can be disabled in mod settings
---------------------------------------------------------------------------------------------------
Version: 0.3.3
Date: 2020.03.24
  Changes:
    - Modified some void recipes to generate 0 pollution: Purest Nitrogen Gas, Alien Bacteria, Phytoplankton, Xyhiphoe Blood, Cold Air, Cold Clean Air, Liquid Nitrogen & Arthropod Blood
---------------------------------------------------------------------------------------------------
Version: 0.3.2
Date: 2020.03.16
  Changes:
    - Increased Ralesia Plantation's pollution absorption from -4/m to -15/m (pyAL only)
    - Reduced Sap Extractor's pollution absorption from -10/m to -5/m (pyAL only)
    - Reduced Moondrop Greenhouse's pollution absorption from -25/m to -20/m (pyAL only)
    - Reduced Moss Farm's pollution absorption from -20/m to -15/m (pyAL only)
    - Reduced Seaweed Crop Plantation's pollution absorption from -20/m to -10/m (pyAL only)
    - Reduced Atomizer's pollution generation from 10/m to 4/m (pyAL only)
    - pyAL added as optional dependency
---------------------------------------------------------------------------------------------------
Version: 0.3.1
Date: 2020.03.09
  Changes:
    - Log recipes no longer scale their absorption modifier with recipe quality
---------------------------------------------------------------------------------------------------
Version: 0.3.0
Date: 2020.03.03
  Info:
    - Added full support for pyAlienLife
  Changes:
    - Rebalanced pollution on factories
    - Most factories have had their pollution/min. reduced
    - Some pollution-absorbing buildings now absorb less
    - Early game mandatory buildings now have much lower pollution/min. (like Destructive Distillation Column) to make pyAL start possible

prototypes/fuels.lua
local function SetFluidFuelData(name, emissions, fuelValue)
	local fuel = data.raw.fluid[name]
	
	if not fuel then
		log("Liquid fuel "..name.." not found.")
		return
	end
	
	if emissions then
		fuel.emissions_multiplier = emissions
	end
	
	if fuelValue then
		fuel.fuel_value = fuelValue
	end
end

local function SetSolidFuelData(name, emissions, fuelValue)
	local fuel = data.raw.item[name]
	
	if not fuel then
		log("Solid fuel "..name.." not found.")
		return
	end
	
	if emissions then
		fuel.fuel_emissions_multiplier = emissions
	end
	
	if fuelValue then
		fuel.fuel_value = fuelValue
	end
end

-- Unprocessed fuels cause more pollution (incentive for processing)
SetSolidFuelData("coal", "1.0", nil)
SetSolidFuelData("raw-coal", "1.05", nil)
SetSolidFuelData("coal-dust", "1.2", nil)
SetSolidFuelData("crushed-coal", "1.05", nil)
SetFluidFuelData("crude-oil", "1.5", nil)
SetFluidFuelData("tar", "1.5", nil)

-- Coke is polluting less than coal, it already generates a lot of pollution during processing anyway
SetSolidFuelData("coke", "0.85", nil)


-- Let's make FuelRods worth the investment
-- Wee need to consider the pollution generated when mining and processing uranium and also when mining and processing other required ores
local fuelRodMk1 = data.raw.item["fuelrod-mk01"]
if fuelRodMk1 then
	fuelRodMk1.fuel_emissions_multiplier = "0.75"
	fuelRodMk1.fuel_acceleration_multiplier = "1.1"
	fuelRodMk1.fuel_top_speed_multiplier = "1.05"
end

local fuelRodMk2 = data.raw.item["fuelrod-mk02"]
if fuelRodMk2 then
	fuelRodMk2.fuel_emissions_multiplier = "0.65"
	--fuelRodMk2.fuel_acceleration_multiplier = "1.2"
	--fuelRodMk2.fuel_top_speed_multiplier = "1.2"
end

local fuelRodMk3 = data.raw.item["fuelrod-mk03"]
if fuelRodMk3 then
	fuelRodMk3.fuel_emissions_multiplier = "0.53"
	--fuelRodMk3.fuel_acceleration_multiplier = "1.5"
	--fuelRodMk3.fuel_top_speed_multiplier = "1.2"
end

local fuelRodMk4 = data.raw.item["fuelrod-mk04"]
if fuelRodMk4 then
	fuelRodMk4.fuel_emissions_multiplier = "0.40"
	--fuelRodMk4.fuel_acceleration_multiplier = "2.0"
	--fuelRodMk4.fuel_top_speed_multiplier = "1.2"
end

local fuelRodMk5 = data.raw.item["fuelrod-mk05"]
if fuelRodMk5 then
	fuelRodMk5.fuel_emissions_multiplier = "0.25"
	--fuelRodMk5.fuel_acceleration_multiplier = "2.3999999999999999"
	--fuelRodMk5.fuel_top_speed_multiplier = "1.2"
end


-- Burning wood produces more pollution than coal
-- https://www.climateinteractive.org/media-coverage/new-york-times-op-ed-burning-wood-is-worse-than-coal-for-the-climate/
SetSolidFuelData("wood", "1.25", nil)
SetSolidFuelData("log", "1.25", nil)
SetSolidFuelData("treated-wood", "1.35", nil) -- It's treated with chemicals! Why would you even do this anyway?
SetSolidFuelData("fiberboard", "1.5", nil)
SetSolidFuelData("formica", "1.6", nil)
SetSolidFuelData("raw-fiber", "1.35", nil)
SetSolidFuelData("fiber", "1.35", nil)


-- Hydrogen has low pollution but needs to be expensive to get (real hydrogen is more expensive to get than the energy it provides)

-- Its energy density is between 120 and 142 MJ/kg
-- https://hypertextbook.com/facts/2005/MichelleFung.shtml

-- A 100% efficient electrolyser requires 39 kWh of electricity to produce 1 kg of hydrogen. The devices today require as much as 48 kWh/kg.
-- http://www.renewableenergyfocus.com/view/3157/hydrogen-production-from-renewables/

-- So it takes 172 MJ to produce 1 kg of Hydrogen with fuel value of ~140 MJ.

-- Electrolyzer MK01 consumes 827 kWh (2977 MJ) to produce 180 000 Hydrogen (in 1 hour) (200 per 4 seconds)
-- TL;DR You need more energy to produce the hydrogen than it provides - usable as "accumulator" since it causes no pollution
SetFluidFuelData("hydrogen", "0.0", "0.015MJ")

-- 500 hydrogen + 600 water + 5 gasoline = 50 pressured hydrogen + 600 Steam (60°C)
SetFluidFuelData("pressured-hydrogen", "0.0", "0.16MJ")


-- "Natural gas emits 50 to 60 percent less carbon dioxide (CO2) when combusted in a new, efficient natural gas power plant compared with emissions from a typical new coal plant [1]."
-- https://www.ucsusa.org/resources/environmental-impacts-natural-gas

SetFluidFuelData("natural-gas", "0.8", "1.0MJ") -- 1MJ
SetFluidFuelData("raw-gas", "0.60", "0.5MJ") -- 0.2MJ
SetFluidFuelData("refined-natural-gas", "0.50", "0.6MJ")-- 0.3MJ
SetFluidFuelData("purified-natural-gas", "0.35", "0.8MJ")-- 0.4MJ
SetFluidFuelData("pure-natural-gas", "0.25", "1.0MJ")-- 0.5MJ


-- The main constituents are, by volume, about 50% hydrogen, 30% methane and 3% higher hydrocarbons, 7% carbon monoxide, 3% carbon dioxide and 7% nitrogen.
-- http://www.thermopedia.com/content/641/
SetFluidFuelData("coke-oven-gas", "0.85", nil)


-- Raw Bitumen is not used as fuel
-- TODO: https://en.wikipedia.org/wiki/Bitumen-based_fuel
-- Remove Bitumen fuel value and add recipe for bitumen fuel (Bitumen + Water + Phenol) or (Bitumen + Water + Some Alcohol (Ethylene?))
local bitumen = data.raw.fluid["bitumen"]
if bitumen then
	bitumen.fuel_value = "0J"
	bitumen.fuel_category = ""
end


-- Syngas, however, has less than half the energy density of natural gas. - this is ok, no fixes
SetFluidFuelData("syngas", "0.9", nil)
SetFluidFuelData("refsyngas", "0.8", nil)


-- Gasoline has lower energy value, but less pollution than diesel
SetFluidFuelData("gasoline", "0.9", nil)

SetFluidFuelData("benzene", nil, "0.8MJ")
SetFluidFuelData("ethylene", "0.95", "1MJ")

SetFluidFuelData("methane", nil, "0.4MJ") -- "0.8MJ" - This is way too easy to get

SetFluidFuelData("propene", "0.90", "1MJ")

SetFluidFuelData("ethanol", "0.90", "0.5MJ")

SetSolidFuelData("biomass", "1.05", "1MJ")
SetSolidFuelData("powdered-biomass", "1.05", nil)
SetSolidFuelData("fine-powdered-biomass", "1.05", nil)
SetSolidFuelData("dried-biomass", "1.05", nil)

SetSolidFuelData("coal-briquette", "1.10", "30MJ")

--Debug
-- for _,v in pairs(data.raw.item) do
	-- if v.fuel_value then
		-- local emissions = (v.fuel_emissions_multiplier or 1) * 100
		-- log("[Solid] "..v.name.." - "..v.fuel_value.."; Emissions: "..emissions.."%")
	-- end
-- end
-- for _,v in pairs(data.raw.fluid) do
	-- if v.fuel_value then
		-- local emissions = v.emissions_multiplier or 1
		-- log("[Fluid] "..v.name.." - "..v.fuel_value.."; Emissions: "..emissions.."%")
	-- end
-- end
